src/components/LoggingExportAppService/LoggingExportAppService.js

```diff
@@ -0,0 +1,167 @@
+ */
+SORT_MESSAGES = false;
+
+/**
+ * Message formatting function which converts each log message from Finsemble's format to the format you wish to transmit.
+ */
+const formatMessage = (logMessage) => {
+	/* Example Incoming log message format:
+	 *		[
+	 *			{
+	 *				"category": "system",				//Log message type: system, dev or perf
+	 *				"logClientName": "Finsemble",		//The registered name of the logger instance
+	 *				"logType": "Log",					//Log level: Error, Warning, Log, Info, Debug, Verbose
+	 *				"logData": "[\"SERVICE LIFECYCLE: STATE CHANGE: Service initializing\",\"windowService\"]",
+	 *													//JSON encoded array of message and data components of the log message
+	 *													//N.B. maybe be prefixed by string "*** Logging Error: ""
+	 *				"logTimestamp": 1544090028391.6226	//Log message timestamp for ordering use
+	 *			},
+	 *			{...},
+	 *			...
+	 *		]
+	 */
+
+	// TODO: add any necessary message format changes here
+	return logMessage;
+};
+
+
+const FSBLReady = () => {
+	try {
+		const batchAllocSize = BATCH_SIZE + 10;   //add a small amount of leeway on batch size
+
+		let timeout = null;
+		let logBatch = new Array(batchAllocSize); //pre-init the batch array to reduce array allocations
+		let currBatchSize = 0;
+
+		/**
+		 * Add an incoming array of log messages to the batch and transmit if thresholds met.
+		 */
+		const addToBatch = (dataArr) => {
+			if (dataArr) {
+				for (let i = 0; i < dataArr.length; i++) {
+					const message = dataArr[i];
+
+					// only capture the log messages that meet the category and type we set
+					if (CAPTURE_LOG_CATEGORIES[message.category] && CAPTURE_LOG_LEVELS[message.logType]) {
+						logBatch[currBatchSize++] = formatMessage(message);
+					}
+					else {
+						// skip the logs that don't meet the requirements
+						console.debug("discarding message", message);
+					}
+				}
+
+				if (currBatchSize >= BATCH_SIZE) {
+					transmitBatch();
+				} else if (!timeout) {
+					// Always transmit batch within timeout
+					timeout = setTimeout(transmitBatch, TIMEOUT_MILLISECS);
+				}
+			} else {
+				Logger.warn("Tried to add an invalid data array to log batch", dataArr);
+			}
+		}
+
+		/**
+		 * Transmit the batch to the remote log collection endpoint.
+		 * @private
+		 */
+		const transmitBatch = () => {
+			clearTimeout(timeout);
+
+			// Trim batch array to length
+			let toTransmit = logBatch;
+			if (toTransmit) {
+				toTransmit.splice(currBatchSize, logBatch.length - currBatchSize);
+			}
+
+			// Reset
+			timeout = null;
+			logBatch = new Array(batchAllocSize);
+			currBatchSize = 0;
+
+			// Sort the batch by timestamp if necessary.
+			if (SORT_MESSAGES) {
+				toTransmit.sort((a, b) => a.logTimestamp - b.logTimestamp);
+			}
+
+			// TODO: Customize batch transmission here
+			// console.debug("Batch to transmit: " + JSON.stringify(toTransmit, null, 2));
```

Comment:
Do we want to move `console.debug` outside of the comment so it actually shows what would be sent to the logging endpoint?